ecs/cloudformation.go

```diff
@@ -34,22 +34,21 @@ import (
 	"github.com/awslabs/goformation/v4/cloudformation/logs"
 	"github.com/awslabs/goformation/v4/cloudformation/secretsmanager"
 	cloudmap "github.com/awslabs/goformation/v4/cloudformation/servicediscovery"
-	"github.com/compose-spec/compose-go/compatibility"
-	"github.com/compose-spec/compose-go/errdefs"
 	"github.com/compose-spec/compose-go/types"
-	"github.com/sirupsen/logrus"
-)
-
-const (
-	parameterClusterName     = "ParameterClusterName"
-	parameterVPCId           = "ParameterVPCId"
-	parameterSubnet1Id       = "ParameterSubnet1Id"
-	parameterSubnet2Id       = "ParameterSubnet2Id"
-	parameterLoadBalancerARN = "ParameterLoadBalancerARN"
 )
 
 func (b *ecsAPIService) Convert(ctx context.Context, project *types.Project) ([]byte, error) {
-	template, networks, err := b.convert(project)
+	err := b.checkCompatibility(project)
+	if err != nil {
+		return nil, err
+	}
+
+	err = b.resources.parse(ctx, project)
```

Comment:
the sequence feels a bit strange here, in basic situation I would expect to get a result from `b.resource.parse()`, and then pass this result to `b.convert()` to get the template. I suppose this was done this way because `convert()` is using some things from the compose project directly, and other from the awsResource created in the `parse()` execution. 
Would it make sense still to make this a bit more explicit with something like 
```
resources, err = b.resources.parse(ctx, project)
if err != nil {
	return nil, err
}

template, err := b.convert(project, resources)
```

or maybe a better way would be to call `b.resource.parse()` as a first step inside `b.parse(project)` rather than requiring both methods to be called in the right order, to limit the statefullness of this

Reply:
`parse` indeed populates `awsResources` from compose file (using x-aws-*)
then `ensure` creates all the infrastructure resources
and eventyally `convert` create the higher level components from the compose model

I'll revisit this, I agree `parse` to return awsResources would be a more logical approach